sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusProcessor.cs

```diff
@@ -222,37 +142,49 @@ public int MaxConcurrentCalls
         ///
         /// <remarks>The message renew can continue for sometime in the background
         /// after completion of message and result in a few false MessageLockLostExceptions temporarily.</remarks>
-        public TimeSpan MaxAutoLockRenewalDuration
-        {
-            get => _maxAutoRenewDuration;
-
-            set
-            {
-                TimeoutHelper.ThrowIfNegativeArgument(value, nameof(value));
-                _maxAutoRenewDuration = value;
-            }
-        }
+        public TimeSpan MaxAutoLockRenewalDuration { get; }
 
         internal bool AutoRenewLock => MaxAutoLockRenewalDuration > TimeSpan.Zero;
 
+        private readonly string _sessionId;
```

Comment:
Don't we need public getter for this?

Reply:
I was on the fence about this since it would only be set if user specifies specific SessionId. I think it would be confusing if it was public for the more common case when this wouldn't be set. The user can get the session Id for the message they are processing from the args.Receiver.SessionManager

Reply:
Yeah...you're right. It would be weird for the non session entities.